tests/sepp/Execution.cpp

```diff
@@ -41,34 +41,63 @@ Execution::Execution(Options const& options, std::shared_ptr<Workload> workload)
 
 Execution::~Execution() {
   _state.store(ExecutionState::kStopped);
+  joinThreads();
+}
+
+void Execution::createThreads(Server& server) {
+  _threads = _workload->createThreads(*this, server);
+}
+
+void Execution::joinThreads() {
   for (auto& thread : _threads) {
     if (thread->_thread.joinable()) {
       thread->_thread.join();
     }
   }
 }
 
-void Execution::createThreads(Server& server) {
-  _threads = _workload->createThreads(*this, server);
-}
-
 ExecutionState Execution::state(std::memory_order order) const noexcept {
   return _state.load(order);
 }
 
+void Execution::signalStartingThread() noexcept { _activeThreads.fetch_add(1); }
 void Execution::signalFinishedThread() noexcept { _activeThreads.fetch_sub(1); }
 
+void Execution::advanceStatusIfNotStopped(ExecutionState state) noexcept {
+  ExecutionState value = _state.load();
+  if (state == ExecutionState::kStopped && value == state) {
+    // already stopped
+    return;
+  }
+  // never move the status backwards
+  while (value != ExecutionState::kStopped &&
+         !_state.compare_exchange_strong(value, state)) {
+    std::this_thread::yield();
+  }
```

Comment:
We can remove the if here. This if is only fulfilled if `value == ExecutionState::kStopped`, but in that case we also won't enter the loop.
```suggestion
  // never move the status backwards
  while (value != ExecutionState::kStopped &&
         !_state.compare_exchange_strong(value, state)) {
    std::this_thread::yield();
  }
```

Reply:
Done